packages/workers/src/build-latest-cache.ts

```diff
@@ -0,0 +1,200 @@
+  } else {
+    const lastProcessedBlockNumber = await getStartBlockNumberFn(db, provider)
+    db.exec(
+      Array.from(watchContractListCheckSum)
+        .map(
+          (address) =>
+            `INSERT OR IGNORE INTO contract_start_block (contract_address, first_block_number) VALUES ('${address}', ${lastProcessedBlockNumber});`,
+        )
+        .join('\n'),
+    )
+  }
+
+  const indexer = new BlockRunner({
+    provider,
+    chainId,
+    getStartBlockNumberFn: () => getStartBlockNumberFn(db, provider),
+    processBlockFn: async (blockNumber) =>
+      processBlockFn({
+        provider,
+        blockNumber,
+        watchContractListLowercase: new Set(
+          Array.from(watchContractListCheckSum).map((address) =>
+            address.toLowerCase(),
+          ),
+        ),
+        db,
+      }),
+    onError: async (latestSafeProcessedBlock: number) =>
+      onError(latestSafeProcessedBlock, db),
+  })
+
+  await indexer.start()
+}
+
+async function getDeFiContractAddressesCheckSum({
+  defiProvider,
+  chainId,
+  chainName,
+}: { defiProvider: DefiProvider; chainId: Chain; chainName: string }) {
+  const defiPoolAddresses = await defiProvider.getSupport({
+    filterChainIds: [chainId],
+  })
+
+  const watchContractList = new Set<string>()
+  for (const pools of Object.values(defiPoolAddresses || {})) {
+    for (const pool of pools) {
+      for (const address of pool.protocolTokenAddresses?.[chainId] || []) {
+        watchContractList.add(getAddress(address).slice(2))
+      }
+    }
+  }
+
+  logger.info(
+    `Watching ${watchContractList.size} DeFi contracts on ${chainName}`,
+  )
+  return watchContractList
+}
+
+async function getStartBlockNumberFn(
+  db: DatabaseType,
+  provider: CustomJsonRpcProvider,
+): Promise<number> {
+  const lastPrcessedBlock = (
+    db
+      .prepare('SELECT latest_block_processed FROM latest_block_processed')
+      .get() as { latest_block_processed?: number } | undefined
+  )?.latest_block_processed
+
+  if (lastPrcessedBlock) {
+    return lastPrcessedBlock
+  }
+
+  const latestBlockNumber = await provider.getBlockNumber()
+
+  return latestBlockNumber
+}
+
+async function processBlockFn({
+  provider,
+  blockNumber,
+  watchContractListLowercase,
+  db,
+}: {
+  provider: CustomJsonRpcProvider
+  blockNumber: number
+  watchContractListLowercase: Set<string>
+  db: DatabaseType
+}): Promise<void> {
+  const receipts = await provider.send('eth_getBlockReceipts', [
+    `0x${ethers.toBeHex(blockNumber).slice(2).replace(/^0+/, '')}`,
+  ])
+
+  const queries: string[] = []
+
+  for (const receipt of receipts?.flat() || []) {
+    for (const log of receipt.logs || []) {
+      // retuned lowercase from provider
+      const contractAddressLowercase = log.address
+      if (watchContractListLowercase.has(contractAddressLowercase.slice(2))) {
+        log.topics
+          .filter((topic: string) =>
+            topic.startsWith('0x000000000000000000000000'),
```

Comment:
Can you add to the filter that `topic !== '0x0000000000000000000000000000000000000000000000000000000000000000'`? We do not want to store references to the zero address in the DB.